src/main/java/org/elasticsearch/action/admin/indices/mapping/get/TransportGetFieldMappingsIndexAction.java

```diff
@@ -133,6 +135,11 @@ protected GetFieldMappingsResponse newResponse() {
         return new GetFieldMappingsResponse();
     }
 
+    @Override
+    protected ClusterBlockException checkRequestBlock(ClusterState state, InternalRequest request) {
+        return state.blocks().indexBlockedException(ClusterBlockLevel.METADATA_READ, request.concreteIndex());
+    }
+
```

Comment:
interesting, so we were previously checking for READ (only data?), which seems like a bug, and now we only look for METADATA_READ blocks?